# This patch is for 770_Better_RSS_Support.diff patch
# It disables the feed header from rss subscription page
# It sends a request to add the rss button into toolbar
# Contributor: Anton Rogaynis
--- mozilla/browser/components/feeds/content/subscribe.xhtml.orig	2008-03-31 16:05:16.000000000 -0400
+++ mozilla/browser/components/feeds/content/subscribe.xhtml	2008-03-31 16:07:45.000000000 -0400
@@ -29,7 +29,8 @@
             src="chrome://browser/content/feeds/subscribe.js"/>
   </head>
   <body onload="SubscribeHandler.writeContent();" onunload="SubscribeHandler.uninit();">
-    <div id="feedHeaderContainer">
+<!--Commenting header from the page to make impossible to subscribe. For subscription we have the button on toolbar -->
+<!--    <div id="feedHeaderContainer">
       <div id="feedHeader" dir="&locale.dir;">
         <div id="feedIntroText">
           <p id="feedSubscriptionInfo1">
@@ -37,10 +38,10 @@
           </p>
           <p id="feedSubscriptionInfo2">&feedSubscriptionInfo2;</p>
         </div>
-
+-->
 <!-- XXXmano this can't have any whitespace in it.  Otherwise you would see
      how much XUL-in-XHTML sucks, see bug 348830 -->
-        <div id="feedSubscribeLine">
+<!--        <div id="feedSubscribeLine">
           <form>
             Subscribe to this feed using
             <select id="handlersMenuList">
@@ -59,7 +60,7 @@
 	</div
       ></div>
     </div>
-
+-->
 	  <!--
 	  <xul:vbox>
             <xul:hbox align="center">
--- mozilla/browser/components/feeds/src/nsFeedSniffer.cpp.orig	2008-03-31 16:05:16.000000000 -0400
+++ mozilla/browser/components/feeds/src/nsFeedSniffer.cpp	2008-03-31 16:10:27.000000000 -0400
@@ -61,6 +61,8 @@
 #include "nsIHttpChannel.h"
 #include "nsIMIMEHeaderParam.h"
 
+#include "nsIObserver.h"
+
 #define TYPE_ATOM "application/atom+xml"
 #define TYPE_RSS "application/rss+xml"
 #define TYPE_MAYBE_FEED "application/vnd.mozilla.maybe.feed"
@@ -387,9 +389,20 @@
   }
 
   // If we sniffed a feed, coerce our internal type
-  if (isFeed && !HasAttachmentDisposition(channel)) 
+  if (isFeed && !HasAttachmentDisposition(channel)) {
     sniffedType.AssignLiteral(TYPE_MAYBE_FEED); 
-  else
+
+    nsCAutoString url;
+    rv = originalURI->GetSpec(url);
+    NS_ENSURE_SUCCESS(rv, PR_TRUE);
+
+    nsCOMPtr<nsISupports> service = do_GetService("@browser/engine-client-observer;1", &rv);
+    NS_ENSURE_SUCCESS(rv, PR_TRUE);
+    nsCOMPtr<nsIObserver> Observer = do_QueryInterface(service, &rv);
+    NS_ENSURE_SUCCESS(rv, PR_TRUE);
+    rv = Observer->Observe(nsnull, "DOMLink_MAYBE_FEED", NS_ConvertUTF8toUTF16(url).get());
+    
+  } else
     sniffedType.Truncate();
   return NS_OK;
 }
